networkit/cpp/graph/Graph.cpp

```diff
@@ -56,7 +56,8 @@ Graph::Graph(std::initializer_list<WeightedEdge> edges) : Graph(0, true) {
     }
 }
 
-Graph::Graph(const Graph &G, bool weighted, bool directed, bool edgesIndexed)
+Graph::Graph(const Graph &G, bool weighted, bool directed, bool edgesIndexed,
+             std::function<edgeweight(edgeweight, edgeweight)> f)
```

Comment:
We can't use `std::function` in performance critical code, `std::function` has a high per-call overhead. Use a template parameter instead.